3.next - Add small and tiny integer types

markstory opened the pull request:
Add `smallint` and `tinyint` types to the schema system. I'm targeting this for 3.5 as I don't think column types changing is acceptable in a bugfix release.

Postgres doesn't support tinyint, so it is mapped to smallint.

Refs #10347

markstory committed:
Add support for tinyint/smallint to MySQL schema

We now support tiny/small int as an abstract type in MySQL driver.

markstory committed:
Add smallint/tinyint types to postgres.

There is no tinyint in postgres, so we'll map it to the smallest int
possible on that platform.

markstory committed:
Add small/tiny int to Sqlite schema.

markstory committed:
Add tiny/small int to Sqlserver.

markstory committed:
Map small/tinyint to number input types.

Treat these types like integer.

markstory committed:
Fix SQLServer tests and add missing coverage.

markstory committed:
Fix failing test in postgres.

inoas commented on src/Database/Schema/SqlserverSchema.php:
```diff
@@ -88,18 +88,18 @@ protected function _convertColumn($col, $length = null, $precision = null, $scal
+            return ['type' => 'tinyint', 'length' => $precision ?: 3];
+        }
+        if ($col === 'smallint') {
+            return ['type' => 'smallint', 'length' => $precision ?: 5];
```
'smallinteger' like 'biginteger' and 'integer'?

inoas commented on src/Database/Schema/SqlserverSchema.php:
```diff
@@ -88,18 +88,18 @@ protected function _convertColumn($col, $length = null, $precision = null, $scal
         }
 
+        if ($col === 'tinyint') {
+            return ['type' => 'tinyint', 'length' => $precision ?: 3];
```
tinyinteger

markstory replied:
That's fair. I'll change it.

sebastienbarre replied:
Hmmm. I can see something coming my way in 2.x :)

markstory replied:
Yes, I've opened a pull request against your fork to update it.

sebastienbarre replied:
Reviewed, thanks. All good except for Sqlite.

inoas reviewed (commented):
Internal type Naming seems to become inconsistent.

markstory committed:
Rename (small|tiny)int to (small|tiny)integer

This makes the naming consistent across integer, biginteger and the new
smaller versions.

markstory committed:
Fix lint error from merge with master.

markstory committed:
Port apcu fixes from 2.x to here.

These fixes helped fix apcu installation issues in 2.x. Hopefully they
do the same here.

markstory committed:
Merge branch '3.next' into 3next-small-tiny-int

markstory committed:
Fix SQLServer test.

markstory commented:
Any other feedback on this before I merge it?

ravage84 commented on src/View/Helper/FormHelper.php:
```diff
@@ -84,10 +84,20 @@ class FormHelper extends Helper
+            'string' => 'text',
+            'datetime' => 'datetime',
+            'boolean' => 'checkbox',
+            'timestamp' => 'datetime',
```
Nitpicking move one up, next to ``datetime``.

ravage84 commented on src/View/Helper/FormHelper.php:
```diff
@@ -84,10 +84,20 @@ class FormHelper extends Helper
+            'datetime' => 'datetime',
+            'boolean' => 'checkbox',
+            'timestamp' => 'datetime',
+            'text' => 'textarea',
```
Nitpicking: Move up to ``string``.

ravage84 commented on src/View/Helper/FormHelper.php:
```diff
@@ -84,10 +84,20 @@ class FormHelper extends Helper
+            'integer' => 'number',
+            'tinyinteger' => 'number',
+            'smallinteger' => 'number',
+            'decimal' => 'number',
```
Nitpicking: Move up to ``float``.

ravage84 reviewed (commented):
Reordering of the type map, would better group the related types together.

markstory committed:
Reorder typemap.

Group related types together.
